packages/gatsby-telemetry/src/__tests__/telemetry.ts

```diff
@@ -57,4 +61,119 @@ describe(`Telemetry`, () => {
       )
     })
   })
+
+  describe(`allows reading tags from path`, () => {
+    it(`getTagsFromPath should read the file and detect updates`, async () => {
+      const t = new AnalyticsTracker({
+        componentId: `component`,
+      })
+
+      // Test it when env not set
+      let res = t.getTagsFromPath()
+      expect(res).toMatchObject({})
+      expect(t.lastEnvTagsFromFileTime).toBe(0)
+
+      // create file and write initial data
+      const filePath = path.join(fs.realpathSync(os.tmpdir()), uuidv4())
+      console.log(filePath)
+      process.env.GATSBY_TELEMETRY_METADATA_PATH = filePath
+
+      fs.writeFileSync(filePath, JSON.stringify({ componentId: `test` }))
+      await new Promise(resolve => {
+        setTimeout(resolve, 2000)
```

Comment:
Why is this necessary?

Reply:
yea, the only way I got it working was to force sleep, I did spend a lot of time trying lower level APIs to ensure I don't get stale reads.